src/create_selection.rs

```diff
@@ -0,0 +1,212 @@
+use core::fmt::{Debug, Display};
+use std::collections::HashSet;
+use std::vec::Vec;
+
+use bdk_coin_select::float::Ordf32;
+use bdk_coin_select::metrics::LowestFee;
+use bdk_coin_select::{
+    Candidate, ChangePolicy, CoinSelector, DrainWeights, FeeRate, NoBnbSolution, Target, TargetFee,
+    TargetOutputs,
+};
+use bitcoin::{Amount, OutPoint, TxOut};
+use miniscript::bitcoin;
+
+use crate::{DefiniteDescriptor, Input, InputGroup, Output};
+
+/// Parameters for creating tx.
+#[derive(Debug, Clone)]
+pub struct CreateSelectionParams {
+    /// All candidate inputs.
+    pub input_candidates: Vec<InputGroup>,
+
+    /// Inputs that must be included in the final tx, given that they exist in `input_candidates`.
+    pub must_spend: HashSet<OutPoint>,
```

Comment:
Use `InputCandidates` here.

Change `InputCandidates` to ensure no duplicates and other checks. I.e. can have `not_canonical` set. Have `must_spends` in `InputCandidates`. `InputCandidates` respects candidate order. Pick `must_spends` first.

Reply:
`finalizer` can be created from `InputCandidates`.

Reply:
Add `bdk_coin_select::Candidate` in `InputCandidates`.